src/values/AbstractObjectValue.js

```diff
@@ -261,6 +279,24 @@ export default class AbstractObjectValue extends AbstractValue {
         return cv.$Get(P, Receiver);
       }
       invariant(false);
+    } else if (this.kind === "conditional") {
+      // this is the join of two concrete objects
```

Comment:
Seems quite redundant with the `$GetOwnProperty` code.

Reply:
It is, but it is also different in subtle ways that make it easier to read in its redundant form. I may come back to this, but at the moment I would prefer to keep things this way.